pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -37,11 +70,308 @@ class LightningGPUStateVector(LightningBaseStateVector):
         device_name(string): state vector device name. Options: ["lightning.gpu"]
     """
 
-    def __init__(self, num_wires, dtype=np.complex128, device_name="lightning.gpu"):
-
-        if device_name != "lightning.gpu":
-            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+    def __init__(
+        self,
+        num_wires,
+        dtype=np.complex128,
+        device_name="lightning.gpu",
+        mpi_handler=None,
+        sync=True,
+    ):
 
         super().__init__(num_wires, dtype)
 
         self._device_name = device_name
+
+        if mpi_handler is None:
+            mpi_handler = LightningGPU_MPIHandler(False, 0, None, num_wires, dtype)
+
+        self._num_global_wires = mpi_handler.num_global_wires
+        self._num_local_wires = mpi_handler.num_local_wires
+
+        self._mpi_handler = mpi_handler
+        self._sync = sync
+
+        # Initialize the state vector
+        if self._mpi_handler.use_mpi:
+            self._qubit_state = self._state_dtype()(
+                self._mpi_handler.mpi_manager,
+                self._mpi_handler.devtag,
+                self._mpi_handler.mpi_buf_size,
+                self._mpi_handler.num_global_wires,
+                self._mpi_handler.num_local_wires,
+            )
+
+        if not self._mpi_handler.use_mpi:
```

Comment:
can we just put it to the else branch here?

Reply:
Done